python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,351 @@
+"""Torch Module for PGExplainer"""
+import math
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.optim import Adam
+
+__all__ = ["PGExplainer"]
+
+
+class PGExplainer(nn.Module):
+    r"""PGExplainer from `Parameterized Explainer for Graph Neural Network
+    <https://arxiv.org/pdf/2011.04573>`
+
+    PGExplainer adopts a deep neural network (explanation network) to parameterize the generation
+    process of explanations, which enables it to explain multiple instances
+    collectively. PGExplainer models the underlying structure as edge
+    distributions, from which the explanatory graph is sampled.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain that tackles multiclass graph classification
+
+        * Its forward function must have the form
+          :attr:`forward(self, graph, nfeat, embed, edge_weight)`.
+        * The output of its forward function is the logits if embed=False else
+          return the intermediate node embeddings.
+    num_features : int
+        Node embedding size used by :attr:`model`.
+    coff_budget : float, optional
+        Size regularization to constrain the explanation size. Default: 0.01.
+    coff_connect : float, optional
+        Entropy regularization to constrain the connectivity of explanation. Default: 5e-4.
+    sample_bias : float, optional
+        Some members of a population are systematically more likely to be selected
+        in a sample than others. Default: 0.0.
+    """
+
+    def __init__(
+        self,
+        model,
+        num_features,
+        coff_budget=0.01,
+        coff_connect=5e-4,
+        sample_bias=0.0,
+    ):
+        super(PGExplainer, self).__init__()
+
+        self.model = model
+        self.num_features = num_features * 2
+
+        # training hyperparameters for PGExplainer
+        self.coff_budget = coff_budget
+        self.coff_connect = coff_connect
+        self.sample_bias = sample_bias
+
+        self.init_bias = 0.0
+
+        # Explanation network in PGExplainer
+        self.elayers = nn.ModuleList()
+        self.elayers.append(
+            nn.Sequential(nn.Linear(self.num_features, 64), nn.ReLU())
+        )
+        self.elayers.append(nn.Linear(64, 1))
```

Comment:
Why not merge L63 - 67 into a single `nn.Sequential`?

Reply:
Done.